File: random_classes/fahrzeug.py
class Fahrzeug():
    alle_marken = ["BMW","Volkswagen","Audi", "Mercedes Benz"]

    def __init__(self, gewicht: int, max_pasagiere: int, max_geschwindigkeit: int, marke: str):
        self.gewicht = gewicht
        self.max_pasagiere = max_pasagiere
        self.max_geschwindigkeit = max_geschwindigkeit
        self.marke = marke
        if self.marke not in Fahrzeug.alle_marken:
            Fahrzeug.alle_marken.append(self.marke)
            Fahrzeug.alle_marken.sort()
        self.motor_zustand = False
        self.geschwindigkeit = 0
    
    def geschwindigkeit_aendern(self, geschwindigkeit=10):
        if self.motor_zustand:
            self.geschwindigkeit += geschwindigkeit
        else:
            return "Der Motor ist Aus."
        
    def gib_geschwindigkeit(self):
        return self.geschwindigkeit
    
    def motor_schalten(self):
        if self.motor_zustand:
            self.motor_zustand = False
        else:
            self.motor_zustand = True
    
    def gib_motor_zustand(self):
        return self.motor_zustand()
    
    def neue_marke_deklarieren(neue_marke: str):
        Fahrzeug.alle_marken.append(neue_marke)

    '''
    def gib_alle_marken():
        for marke in Fahrzeug.alle_marken():
            print(marke)
    '''
        

File: random_classes/FAHRZEUG.java
public class FAHRZEUGE
{
   // Um für die Subklassen zugänglich zu machen, müssen die Variablen protected oder public sein, nicht private.
   int gewicht; // in Tonnen
   int pasagiere;
   int bauJahr;
   
   int maxGeschwindigkeit;
   int leistung;
   int anzahlRaeder;
   
   int anzahlSitze;
   
   
   int currentGeschwindigkeit;
   String zustand;
   
   //     FAHRZEUGE(Gewicht,     Pasagiere,      bauJahr,      maximale Geschwindigkeit,   leistung,    Anzahl der Räder,    Anzahl der Sitze,  Markenname)
   public FAHRZEUGE(int gewicht1, int pasagiere1, int bauJahr1, 
                   int maxGeschwindigkeit1, int leistung1, int anzahlRaeder1, 
                   int anzahlSitze1)
   {
       currentGeschwindigkeit = 0;
       zustand = "Aus";
       gewicht = gewicht1;
       pasagiere = pasagiere1;
       bauJahr = bauJahr1;
       maxGeschwindigkeit = maxGeschwindigkeit1;
       leistung = leistung1;
       anzahlRaeder = anzahlRaeder1;
       anzahlSitze = anzahlSitze1;
   }
   
   public void geschwindigkeitAendern(int zielGeschwindigkeit){
       if (zustand == "An") {
           currentGeschwindigkeit = zielGeschwindigkeit;
       }
       else {
           return;
       }
   }
   
   public String startUndAusKnopf() {
       if (zustand == "Aus") {
           zustand = "An";
           return "Der Motor ist An";
       }
       else {
           zustand = "Aus";
           currentGeschwindigkeit = 0;
           return "Der Motor ist Aus.";
       }
   }
   
   public int gibCurrentGeschwindigkeit() {
       return currentGeschwindigkeit;
   }
   
}
